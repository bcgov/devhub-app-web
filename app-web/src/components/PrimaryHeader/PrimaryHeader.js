import React from 'react';
import PropTypes from 'prop-types';
import Banner from '../UI/Banner/Banner';
import Hamburger from '../UI/Hamburger/Hamburger';
// login not being implemented at this time
import Login from '../Auth/Login';
import styled from '@emotion/styled';
import { CUSTOM_BREAKPOINTS } from '../../constants/designTokens';

const Header = styled.header`
  background-color: ${({ theme }) => theme.colors.blue};
  border-bottom: 2px solid ${({ theme }) => theme.colors.yellow};
  padding: 0 12px 0 25px;
  color: #fff;
  display: flex;
  box-shadow: 0 1px 2px 2px rgba(0, 0, 0, 0.25);
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  z-index: 50;
  height: 65px;
  @media screen and (min-width: 600px) {
    padding: 0 65px 0 65px;
  }
`;

const StyledHamburger = styled(Hamburger)`
  ${CUSTOM_BREAKPOINTS.navbar} {
    display: none;
  }
`;

const LoginWrapper = styled.div`
  display: none;

  ${CUSTOM_BREAKPOINTS.navbar} {
    display: inline-block;
  }
`;

const Panel = styled.div`
  flex-grow: 1;
  align-items: center;
  justify-content: flex-end;
  display: flex;
`;

export const PrimaryHeader = ({ showHamburger, hamburgerClicked }) => {
  return (
    <Header>
      <Banner />
      <Panel>
        <LoginWrapper>
          <Login />
        </LoginWrapper>
        {showHamburger ? <StyledHamburger clicked={hamburgerClicked} /> : null}
      </Panel>
    </Header>
  );
};

PrimaryHeader.propTypes = {
  showHamburger: PropTypes.bool,
  hamburgerClicked: PropTypes.func,
};

PrimaryHeader.defaultProps = {
  showHamburger: false,
  hamburgerClicked: () => undefined,
};

export default PrimaryHeader;
