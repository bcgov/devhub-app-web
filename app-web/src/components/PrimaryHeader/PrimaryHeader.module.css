.PrimaryHeader {
    background-color: #036;
    border-bottom: 2px solid #fcba19;
    padding: 0 25px;
    color: #fff;
    display: flex;
    height: 65px;
    box-shadow: 0 1px 2px 2px rgba(0, 0, 0, .25);
    position: relative;
}


@media screen and (min-width: 600px) {
    .PrimaryHeader {
        padding: 0 65px 0 65px;
    }
   
}
  