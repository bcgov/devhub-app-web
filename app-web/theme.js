/*
Copyright 2019 Province of British Columbia

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at 

   http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

Created by Patrick Simonian
*/

/*
Emotion theming object

We are transitioning from common elements such as colors, background colors being abstracted away into
a themes provider. This is a very common practice now a days and makes it maintain themes on the fly
docs on theming is here https://emotion.sh/docs/theming#___gatsby

Docs on emotion https://emotion.sh/docs/introduction

--- our idea is to introduce emotion early and slowly replace css modules with emotion styled components
when and as needed. This is a slow processes but will lead to better and more maintaible styling in the long run.
*/

import { SPACING } from './src/constants/designTokens';
import { RESOURCE_TYPES, TOPICS, SEARCH_RESOURCE_TYPES, JOURNEY } from './src/constants/ui';

export const DEVHUB_PALETTE = {
  [RESOURCE_TYPES.COMPONENTS]: '#4299D2',
  [RESOURCE_TYPES.DOCUMENTATION]: '#246BD1',
  [RESOURCE_TYPES.PEOPLE]: '#E17039',
  [RESOURCE_TYPES.SELF_SERVICE_TOOLS]: '#F3BA45',
  [TOPICS]: '#444',
  [RESOURCE_TYPES.REPOSITORIES]: '#6e5494',
  [SEARCH_RESOURCE_TYPES.GITHUB_ISSUE]: '#6e5494',
  [RESOURCE_TYPES.EVENTS]: '#f44b42',
  [JOURNEY]: '#009688',
  lightgreen: '#66bb6a',
  yellow: '#FCBA19',
  lightyellow: '#F3BA45',
  grey: '#494949',
  lightgrey: '#F2F2F2',
  darkgrey: '#444',
  orange: '#E17039',
  purple: '#6E5494',
  red: '#D8292F',
  lightred: '#F44B42',
  green: '#2E8540',
  blue: '#003366',
  babyblue: '#4299D2',
  lightblue: '#246BD1',
  linkblue: '#1A5A96',
  bgBlue: '#38598A',
  white: '#ffffff',
};

export default {
  colors: {
    ...DEVHUB_PALETTE,
  },
  padding: {
    main: {
      mobile: `${SPACING['2x']} ${SPACING['2x']}`,
      desktop: `${SPACING['2x']} ${SPACING['13x']}`,
    },
  },
  breakpoints: {
    // specific breakpoints
    main: {
      desktop: '@media screen and (min-width: 932px)',
    },
  },
};
