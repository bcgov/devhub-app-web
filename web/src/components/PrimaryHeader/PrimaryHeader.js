import React from 'react';
import PropTypes from 'prop-types';
import classes from './PrimaryHeader.module.css';
import Banner from '../UI/Banner/Banner';
import Hamburger from '../UI/Hamburger/Hamburger';
// login not being implemented at this time
// import Login from '../Auth/Login/Login';

export const PrimaryHeader = ({ showHamburger, hamburgerClicked }) => (
  <header className={classes.PrimaryHeader}>
    <Banner />
    <div className={classes.Other}>
      {/* <Login /> */}
      {showHamburger ? (
        <Hamburger clicked={hamburgerClicked} className={classes.Hamburger} />
      ) : null}
    </div>
  </header>
);

PrimaryHeader.propTypes = {
  showHamburger: PropTypes.bool,
  hamburgerClicked: PropTypes.func,
};

PrimaryHeader.defaultProps = {
  showHamburger: false,
  hamburgerClicked: () => undefined,
};

export default PrimaryHeader;
