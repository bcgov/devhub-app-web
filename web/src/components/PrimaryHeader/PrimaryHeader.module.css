.PrimaryHeader {
    background-color: #036;
    border-bottom: 2px solid #fcba19;
    padding: 0 12px 0 25px;
    color: #fff;
    display: flex;
    box-shadow: 0 1px 2px 2px rgba(0, 0, 0, .25);
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
    z-index: 50;
    height: 65px;
}

.Other {
    flex-grow: 1;
    align-items: center;
    justify-content: flex-end;
    display: flex;
}

.PrimaryHeader .Hamburger {
    cursor: pointer;
}

@media screen and (min-width: 600px) {
    .PrimaryHeader {
        padding: 0 65px 0 65px;
    }
}

@media screen and (min-width: 932px) {
    .PrimaryHeader .Hamburger {
        display: none;
    }
}
