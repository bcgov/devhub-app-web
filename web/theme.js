/*
Copyright 2019 Province of British Columbia

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at 

   http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

Created by Patrick Simonian
*/

/*
Emotion theming object

We are transitioning from common elements such as colors, background colors being abstracted away into
a themes provider. This is a very common practice now a days and makes it maintain themes on the fly
docs on theming is here https://emotion.sh/docs/theming#___gatsby

Docs on emotion https://emotion.sh/docs/introduction

--- our idea is to introduce emotion early and slowly replace css modules with emotion styled components
when and as needed. This is a slow processes but will lead to better and more maintaible styling in the long run.
*/

import { SPACING, DEVHUB_PALETTE } from './src/constants/designTokens';
export default {
  colors: {
    ...DEVHUB_PALETTE,
  },
  padding: {
    main: {
      mobile: `${SPACING['2x']} ${SPACING['2x']}`,
      desktop: `${SPACING['2x']} ${SPACING['13x']}`,
    },
  },
  breakpoints: {
    // specific breakpoints
    main: {
      desktop: '@media screen and (min-width: 932px)',
    },
  },
};
