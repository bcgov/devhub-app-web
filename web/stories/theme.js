import { create } from '@storybook/theming';
import { DEVHUB_PALETTE } from '../theme';

export default create({
  base: 'light',

  colorPrimary: 'hotpink',
  colorSecondary: 'deepskyblue',

  // UI
  appBg: 'white',
  appContentBg: 'white',
  appBorderColor: 'grey',
  appBorderRadius: 4,

  // Typography
  fontBase: 'sans-serif',
  fontCode: 'monospace',

  // Text colors
  textColor: DEVHUB_PALETTE.darkgrey,
  textInverseColor: 'rgba(255,255,255,0.9)',

  // Toolbar default and active colors
  barTextColor: DEVHUB_PALETTE.white,
  barSelectedColor: 'black',
  barBg: DEVHUB_PALETTE.blue,

  // Form colors
  inputBg: 'white',
  inputBorder: 'silver',
  inputTextColor: 'black',
  inputBorderRadius: 4,

  brandTitle: 'My custom storybook',
  brandImage:
    'https://u58tqm22fk-flywheel.netdna-ssl.com/wp-content/uploads/2016/06/BC-Gov-Logo-CMYK_pos.png',
});
