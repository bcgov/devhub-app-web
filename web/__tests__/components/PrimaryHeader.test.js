import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from '@emotion/react';
import theme from '../../theme';
import { PrimaryHeader } from '../../src/components/PrimaryHeader/PrimaryHeader';

describe('Primary Header Component', () => {
  it('matches snapshot', () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <PrimaryHeader />
      </ThemeProvider>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
