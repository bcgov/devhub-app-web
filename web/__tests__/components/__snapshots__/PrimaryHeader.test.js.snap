// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Primary Header Component matches snapshot 1`] = `
.emotion-19 {
  background-color: #003366;
  border-bottom: 2px solid #FCBA19;
  padding: 0 12px 0 25px;
  color: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-shadow: 0 1px 2px 2px rgba(0,0,0,0.25);
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  z-index: 50;
  height: 65px;
}

@media screen and (min-width:600px) {
  .emotion-19 {
    padding: 0 65px 0 65px;
  }
}

.emotion-11 {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  color: #1A5A96;
  font-weight: 400;
  text-transform: capitalize;
  padding: 0 4px;
  margin: 0 2px;
  color: #fff;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: inherit;
  margin: 0 10px 0 0;
}

.emotion-11:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
}

.emotion-11:visited {
  color: inherit;
}

.emotion-0 {
  margin-bottom: 0;
  width: 175px;
}

@media (max-width:480px) {
  .emotion-0 {
    display: none;
  }
}

.emotion-2 {
  margin-bottom: 0;
  width: 50px;
}

@media (min-width:480px) {
  .emotion-2 {
    display: none;
  }
}

.emotion-8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 1em;
}

.emotion-4 {
  margin: 6px 3px 6px 0;
  padding: 0px 4px;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1.54912em;
}

.emotion-6 {
  text-transform: uppercase;
  font-weight: 600;
  color: #fcba19;
}

.emotion-17 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-15 {
  display: none;
}

@media (min-width:932px) {
  .emotion-15 {
    display: inline-block;
  }
}

.emotion-13 {
  color: #fff;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  border: 1px solid transparent;
  padding: 0.375rem 0.75rem;
  font-size: 1em;
  line-height: 1.5;
  border-radius: 0.25rem;
  cursor: pointer;
  font-weight: 400;
  text-align: center;
  white-space: nowrap;
  vertical-align: middle;
  display: inline-block;
  background-color: #FCBA19;
}

<header
  class="emotion-19 emotion-20"
>
  <a
    class="emotion-10 emotion-11 emotion-12"
    data-testid="header.govlogo.banner"
    href="/"
  >
    <img
      alt="Government of British Columbia"
      class="emotion-0 emotion-1"
      src="BCID_H_rgb_rev.svg"
    />
    <img
      alt="Government of British Columbia"
      class="emotion-2 emotion-3"
      src="logo.svg"
    />
    <div
      class="emotion-8 emotion-9"
    >
      <h2
        class="emotion-4 emotion-5"
      >
        DevHub
      </h2>
      <div
        aria-label="This application is currently in beta phase"
        class="emotion-6 emotion-7"
      >
        beta
      </div>
    </div>
  </a>
  <div
    class="emotion-17 emotion-18"
  >
    <div
      class="emotion-15 emotion-16"
    >
      <button
        class="emotion-13 emotion-14"
        color="yellow"
        data-testid="auth.login"
      >
        Login
      </button>
    </div>
  </div>
</header>
`;
